src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResourceIntTest.kt

```diff
@@ -179,91 +179,192 @@ class ActivitiesResourceIntTest : IntegrationTestBase() {
   @Nested
   inner class UpdateActivity {
 
-    private fun updateActivityRequest() = UpdateActivityRequest(prisonId = "LEI")
+    private fun updateActivityRequest() = UpdateActivityRequest(
+      prisonId = "LEI",
+      code = "CA",
+      programCode = "SOME_PROGRAM",
+      description = "test description",
+      capacity = 12,
+      startDate = LocalDate.of(2022, 10, 31),
+      endDate = LocalDate.of(2022, 11, 30),
+      minimumIncentiveLevelCode = "STD",
+      internalLocationId = -27,
+      payRates = listOf(PayRateRequest(incentiveLevel = "BAS", payBand = "5", rate = BigDecimal.valueOf(0.4)))
+    )
 
-    @Test
-    fun `access forbidden when no authority`() {
-      webTestClient.put().uri("/activities/1")
-        .contentType(MediaType.APPLICATION_JSON)
-        .body(BodyInserters.fromValue(updateActivityRequest()))
-        .exchange()
-        .expectStatus().isUnauthorized
-    }
+    private fun updateActivityRequestJson(
+      prisonIdJson: String? = """"prisonId": "LEI",""",
+      capacityJson: String = """"capacity": 12,""",
+      startDateJson: String = """"startDate" : "2022-10-31",""",
+      payRatesJson: String? = """
+          "payRates" : [ {
+              "incentiveLevel" : "BAS",
+              "payBand" : "5",
+              "rate" : 0.4
+              } ]
+      """.trimIndent(),
+    ) = """{
+            ${prisonIdJson ?: ""}
+            "code" : "CA",
+            "programCode" : "SOME_PROGRAM",
+            "description" : "test description",
+            $capacityJson
+            $startDateJson
+            "endDate" : "2022-11-30",
+            "minimumIncentiveLevelCode" : "STD",
+            "internalLocationId" : -27,
+            ${payRatesJson ?: ""}
+          }"""
 
-    @Test
-    fun `access forbidden when no role`() {
-      webTestClient.put().uri("/activities/1")
-        .contentType(MediaType.APPLICATION_JSON)
-        .headers(setAuthorisation(roles = listOf()))
-        .body(BodyInserters.fromValue(updateActivityRequest()))
-        .exchange()
-        .expectStatus().isForbidden
-    }
+    @Nested
+    inner class Api {
```

Comment:
Existing tests haven't changed apart from to move inside this class which has apparenly confused GitHub's diff engine 🙄